toytools/transform.py

```diff
@@ -72,3 +72,68 @@ def try_find_region_with_signal(
 
     return (cropped_image, crop_region)
 
+
+#===================================================================================
+# ============================= Find Multi-track windows ===========================
+def multitrack_preprocess(x, kernel_size=2, stride=2):
+	"""
+	Preprocessing steps for multitrack detections.
+	Input:
+		- x (numpy.ndarray): the input image of shape (H, W)
+		- kernel_size (int): the kernel size of torch.AvgPool2d
+		- stride (int): the stride of torch.AvgPool2d
+	Output (numpy.ndarray): the processed 0-1 image
+	"""
+    x = np.abs(x)
+    x[x > 0] = 1
+    with torch.no_grad():
+        x = torch.tensor(x).to(torch.float)
+        input = x.view(1, 1, x.shape[0], x.shape[1])
+        avgpool = nn.AvgPool2d(kernel_size, stride=stride)
```

Comment:
Hi Dmitrii, occasionally (and rarely), even when there is a single track, there can be a zero streak in the middle of a track (probably since the data is bipolar), in which case the scan will think there are multi-tracks. But it is probably really really rare. I use pooling to average things out so that the zero will disappear. But I agree that it is probably not very necessary.

Reply:
This is a good point.  And, next sample when we add "real" Geant4 stepping through the LAr medium the tracks will reflect dE/dx fluctuations, multiple scattering, delta ray emissions, and more.  All these will cause pixels to dip below threshold when viewing just 1 view of the 2D image even though the "peaks" on either side of the "valley" might be considered connected.

IOW, `is_multitrack()` is a hard function to implement in general!

But, it is still practical to have a simple function even if "multitrack" is maybe open to different interpretations.

Reply:
Thank you for the explanation, Yi!